File: Project1/Java/src/Main.java
import java.io.*;
import java.util.Scanner;

public class Main
{
    public static void main(String[] args) 
    {

        try 
        {
            Scanner reader = new Scanner(new FileReader(args[0]));
            FileWriter writer = new FileWriter(args[1]);
            
            LinkedList ll = new LinkedList();
            System.out.println(ll);
            writer.write(ll.toString() + "\n");

            String s;
            while (reader.hasNext())
            {
                s = reader.next();
                ll.listInsert(s);
                ll.printList();
                writer.write(ll.toString() + "\n");
            }
   
            reader.close();
            writer.close();
        }
        catch(Exception e)
        {
            printError(e);
        }

    }

    // For debugging purposes.
    public static void printError(Exception e) 
    {
        StringWriter writer = new StringWriter();
        PrintWriter printWriter = new PrintWriter(writer);
        e.printStackTrace(printWriter);
        printWriter.flush();
        String stackTrace = writer.toString();
        System.out.println(stackTrace);
    }
}

File: Project1/C++/src/Main.cpp
#include "LinkedList.cpp"
#include <fstream>

int main(int argc, char** argv)
{
    LinkedList* ll = new LinkedList();
    ifstream input;
    ofstream output;

    input.open(argv[1]);
    output.open(argv[2]);
    ll->printList();
    output << ll->toString() << endl;
    string word;
    int i = 0;

    while (input.good()) 
    {
        input >> word;        
        ll->listInsert(word);
        if (i < 15)
        {
            output << ll->toString() << endl;
            ll->printList();
            i++;
        }
    }

    output.close();
    input.close();
    return 0;
}